README.md

```diff
@@ -15,28 +15,51 @@ Examples
 
 ```erlang
 {ok, Bin} = file:read_file("test/doc.json").
-Doc = jiffy:decode(Bin, [return_maps]).
+Doc = jsx:decode(Bin, [return_maps]).
 
 %% return 1'st book author
-[<<"Nigel Rees">>] = ejsonpath:execute("$.store.book[0].author", Doc).
+{[<<"Nigel Rees">>], ["$['store']['book'][0]['author']"]} = ejsonpath:q("$.store.book[0].author", Doc).
 
 %% return 1'st book categody and author
-[<<"reference">>,
- <<"Nigel Rees">>] = ejsonpath:execute("$.store.book[0]['category','author']", Doc).
+{[<<"reference">>,<<"Nigel Rees">>],
+ ["$['store']['book'][0]['category']",
+  "$['store']['book'][0]['author']"]} = ejsonpath:q("$.store.book[0]['category','author']", Doc).
 
 %% return only reference book authors
 %% `Funs' is a map or propist of `Name` and `Fun` pairs (see Fun spec in the sources)
 Funs = #{
- <<"filter_category">> =>
+ filter_category =>
   fun({Map, _Doc}, [CategoryName]) ->
      case maps:find(<<"category">>, Map) of
          {ok, CategoryName} -> true;
          _ -> false
      end
   end
 },
-[<<"Nigel Rees">>] = ejsonpath:execute(
-                     "$.store.book[?(filter_category('reference'))].author", Doc, Funs).
+{[<<"Nigel Rees">>],["$['store']['book'][0]['author']"]} = ejsonpath:q("$.store.book[?(filter_category('reference'))].author", Doc, Funs).
+
+%% update only item id == 0
+O = #{
+  <<"items">> => [
+    #{<<"id">> => 0, <<"value">> => yyy},
+    #{<<"id">> => 1, <<"value">> => yyy}
+  ]
+},
+
+ejsonpath:tr("$.items[?(@.id == 0)].value", O, fun(_) -> xxx end).
+{
+  #{<<"items">> =>
+       [#{<<"id">> => 0,<<"value">> => xxx},
+        #{<<"id">> => 1,<<"value">> => yyy}]},
+  ["$['items'][0]['value']"]
+ 
+
+% delete item if query match
+ejsonpath_transform:tr("$.items[?(@.id == 0)]", O, fun (_) -> delete end, #{}, []))
```

Comment:
and would be nice to explain what are the last 2 arguments for?

And other possible return values for `TransformFun`:
```
{error, Error}
{ok, Result}
```